[Benjamin Tay] ip

Btaykb opened the pull request:
### **Duke 2.0**

> The new and improved Duke app that improves on the original!

- Keep tracks and manages _**all your tasks**_
- _**Easy**_ to learn 
- _**New features**_ continuously added 

Use Duke in just these simple steps 😄 
1.  Download the latest version [here](https://github.com/Btaykb/ip/releases)
2.  Run `java -jar `  followed by the file path of the jar file
3.  Duke is ready to go!

Features: 
- [X] Add to-do, deadline and event type tasks
- [X] Mark or unmark current tasks
- [X] Delete any tasks in your list
- [X] Search for tasks using keywords

Calling all Java Programmers! Your own custom features can be added in due to the short and simple methods in Duke!
Here's the `main` method: 
``` java
public static void main(String[] args) throws IOException, DukeException {
        new Duke().run();
}
```

Commit:
Add Gradle support

Btaykb committed:
completed Level-1

Btaykb committed:
completed Level-2

Btaykb committed:
added Task class and completed Level-3

Btaykb committed:
completed Level-4

Btaykb committed:
completed A-TextUiTesting

Btaykb committed:
completed Level-5

Btaykb committed:
completed Level-6

Btaykb committed:
add save support for task creation, mark, unmark and delete

Btaykb committed:
add date formatting for deadline tasks

Btaykb committed:
complete A-MoreOOP, added Ui, Storage, Parser, Tasklist, Command classes

Btaykb committed:
complete A-Packages, add suitable java packages

Btaykb committed:
complete A-JUnit, add TodoTest, DeadlineTest, ParserTest

Btaykb committed:
complete A-JavaDoc, add JavaDoc to all files

Btaykb committed:
complete A-CodingStandard

Btaykb committed:
complete Level-9, add find feature

Btaykb committed:
fix merge conflict in InvalidCommand

Btaykb committed:
fix merge conflict in the data files

Btaykb committed:
add ClearCommand and fixed some coding standard issues

Btaykb committed:
fixed some names

Btaykb committed:
Merge branch 'add-gradle-support'

Btaykb committed:
add gui for duke

lzf834 commented on src/main/java/duke/FileHandler.java:
```diff
@@ -0,0 +1,143 @@
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
I like how you separated all the different imports according to how and when they were used.

lzf834 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,26 @@
+delete -1
+delete 9 8 2
+delete 100
+bye
```
Good job with the code, I had an easy time as your code followed the Java coding standard to the tee. Keep it up!

lzf834 reviewed (commented):
Overall Fantastic

wxliong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,101 @@
+        return false;
+    }
+
+    /**
+     * Will add the provided task to the tasklist.
+     *
+     * @param stg The storage object to use file writing methods.
+     * @param ui The ui object to handle I/O requests.
+     * @param tasks The task list which holds all tasks available.
+     * @throws IOException If an I/O error occurs.
+     */
+    @Override
+    public String execute(Storage stg, Ui ui, TaskList tasks) throws IOException {
```
According to [our coding standard](https://se-education.org/guides/conventions/java/intermediate.html), 

> In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)

wxliong commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+ */
+public abstract class Command {
+
+    /**
+     * Based on the Command type, will execute its appropriate
+     * action.
+     *
+     * @param stg The storage object to use file writing methods.
+     * @param ui The ui object to handle I/O requests.
+     * @param tasks The task list which holds all tasks available.
+     * @throws DukeException If any error related to the Duke app occurs.
+     * @throws IOException If an I/O error occurs.
+     */
+    public abstract String execute(Storage stg, Ui ui, TaskList tasks) throws DukeException, IOException;
```
Based on the coding standards, it seems like your Javadoc is missing the @return statement.

wxliong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,70 @@
+     * @return A Command object that will carry out the appropriate
+     * action.
+     */
+    public Command parse(String text) {
+        String input = text.trim();
+        String command = getCommand(input);
+        if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.equals("mark") || command.equals("unmark")
+                || command.equals("delete")) {
+            return new NumberedCommand(getIndex(input), command, input);
+        } else if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("todo") || command.equals("deadline")
+                || command.equals("event")) {
+            return new AddCommand(command, input);
+        } else if (command.equals("find")) {
+            return new FindCommand(input);
+        } else if (command.equals("clear")) {
+            return new ClearCommand();
+        } else {
+            return new InvalidCommand();
+        }
+    }
```
Perhaps using switch here may help with the readability of the code? Also, using "fall through" for commands like mark/unmark :)
An example can be found [here](https://se-education.org/guides/conventions/java/intermediate.html) under switch statement.

wxliong reviewed (commented):
A few minor suggestions but overall, good and readable code! I like how you did your abstraction :)

Btaykb committed:
improved gui

Btaykb committed:
complete A-Assertions, add assertions

Btaykb committed:
complete A-CodeQuality

Btaykb committed:
Merge pull request #1 from Btaykb/branch-A-Assertions

complete A-Assertions

Btaykb committed:
Merge pull request #2 from Btaykb/branch-A-CodeQuality

complete A-CodeQuality

Btaykb committed:
Add checkstyle support

Added checkstyle support for Duke app. Users are now able to check
and see if their code is compliant with the coding standards specified.
Rather than manually having to read through each line to detect any
coding violations.

checkstyle.xml and suppression.xml were added in ./config/checkstyle.

Btaykb committed:
Code Formatting

*Added function to filter TaskList so as to tidy FindCommand.java
*Added missing JavaDocs for FileHandler and TaskList

Btaykb committed:
Add undo feature

Btaykb committed:
Improved GUI, made chat box fit the size of words used

Btaykb committed:
Reduce method length for AddCommand, UndoCommand, NumberedCommand

Btaykb committed:
Update User Guide

Btaykb committed:
Improve User Guide

Btaykb committed:
Set theme jekyll-theme-cayman

Btaykb committed:
Set theme jekyll-theme-minimal

Btaykb committed:
Fix bug with no index input case for mark, unmark and delete.

Btaykb committed:
Merge branch 'master' of https://github.com/Btaykb/ip

Btaykb committed:
Fix bug for file accessing
